syntax = "proto3";

service LamodaService {
    rpc AppendTagsToCategoryCSV (TagsRequest) returns (Response);
    rpc SaveImagesToCategory (ImageRequest) returns (Response);
    rpc CreateCategoryDirectory (CategoryRequest) returns (Response);
}

message TagsRequest {
    repeated string tags = 1;
    string image_path = 2;
    string category_name = 3;
}

message ImageRequest {
    bytes image_data = 1;
    string image_name = 2;
    string category_name = 3;
}

message CategoryRequest {
    string category_name = 1;
    string base_dir = 2;
}

message Response {
    bool success = 1;
    string message = 2;
}
